src/routes/MainPage.jsx

```diff
@@ -1,41 +1,61 @@
 import { useDispatch, useSelector } from 'react-redux';
 import { useEffect } from 'react';
+import groupingDoctors from '../logics/groupingDoctors';
 import {
   getMainPageDoctors,
   mainPageDoctors,
 } from '../redux/mainPage/mainPageSlice';
 import { fetchAllDoctors } from '../redux/doctor/doctorSlice';
-import DoctorCard from '../components/DoctorCard';
+import MainPageCaroussel from '../components/MainPageCaroussel';
 
 const MainPage = () => {
   const dispatch = useDispatch();
   const { doctors, isLoading, error } = useSelector(mainPageDoctors);
+  let allDoctors = useSelector((state) => state.doctor.doctors);
 
   useEffect(() => {
-    dispatch(getMainPageDoctors());
-    dispatch(fetchAllDoctors());
+    if (doctors.length === 0) dispatch(getMainPageDoctors());
+    if (allDoctors.length === 0) dispatch(fetchAllDoctors());
```

Comment:
No need for this as `fetchAllDoctors` is being called in `Layout.jsx` (ref my PR)